low-level/files/opt/seagate/sspl/setup/resource_map/server.py

```diff
@@ -22,28 +22,220 @@
 
     def __init__(self):
-        """Initialize server"""
-        pass
+        """Initialize server."""
+        super().__init__()
+        self.server_frus = {
+            'sas_hba': self.get_sas_hba_info,
+            'sas_ports': self.get_sas_ports_info,
+            'nw_ports': self.get_nw_ports_info
+        }
+        self.sysfs = ToolFactory().get_instance('sysfs')
+        self.sysfs.initialize()
 
     def get_health_info(self, rpath):
         """
-        Get health information of fru in given resource id
+        Get health information of fru in given resource id.
 
         rpath: Resource id (Example: nodes[0]>compute[0]>hw>disks)
         """
-        pass
+        info = {}
+        fru_found = False
+        nodes = rpath.strip().split(">")
+        leaf_node, _ = self.get_node_details(nodes[-1])
+        if leaf_node == "compute":
+            for fru in self.server_frus:
+                info.update({fru: self.server_frus[fru]()})
+            info["last_updated"] = int(time.time())
+            fru_found = True
+        else:
+            fru = None
+            for node in nodes:
+                fru, _ = self.get_node_details(node)
+                if self.server_frus.get(fru):
+                    fru_found = True
+                    info = self.server_frus[fru]()
+                    break
+        if not fru_found:
+            raise ResourceMapError(
+                errno.EINVAL,
+                "Health provider doesn't have support for'{rpath}'.")
+        return info
+
+    @staticmethod
+    def get_node_details(node):
+        """
+        Parse node information and returns left string and instance.
+
+        Example
+            "storage"    -> ("storage", "*")
+            "storage[0]" -> ("storage", "0")
+        """
+        res = re.search(r"(\w+)\[([\d]+)\]|(\w+)", node)
+        inst = res.groups()[1] if res.groups()[1] else "*"
+        node = res.groups()[0] if res.groups()[1] else res.groups()[2]
+        return node, inst
+
+    def get_data_template(self, uid, is_fru: bool):
+        """Get the health data template."""
+        return {
+            "uid": uid,
+            "fru": str(is_fru).lower(),
+            "last_updated": int(time.time()),
+            "health": {
+                "status": "",
+                "description": "",
+                "recommendation": "",
+                "specifics": []
+            }
+        }
+
+    def set_health_data(self, obj: dict, status, description=None,
+                        recommendation=None):
+        """Set the given or default health data as per health status."""
+        good_state = (status == "OK")
+        if not description:
+            description = "%s is %sin good state" % (
+                            obj.get("uid"),
+                            '' if good_state else 'not ')
+        if not recommendation:
+            recommendation = 'None' if good_state\
+                             else "Contact Seagate Support."
```

Comment:
default recommendation to maintain & referred from sspl-constants

Reply:
done.